martian/src/mro.rs

```diff
@@ -181,14 +182,16 @@ mro_display_to_display! {MartianPrimaryType}
 #[derive(Debug, Serialize, Deserialize, Clone, PartialEq, Eq, Hash)]
 pub enum MartianBlanketType {
     Primary(MartianPrimaryType),
-    Array(MartianPrimaryType),
+    Array(Box<MartianBlanketType>),
+    TypedMap(Box<MartianBlanketType>),
```

Comment:
Can `TypedMap` be a `MartianPrimaryType`?

Reply:
Hmmm what would be the benefit of that over this design?

Reply:
Even if it were a `MartianPrimaryType` TypedMap would still need to contain `Box<MartianBlanketType>` if we want it to be able to contain `Array`s.

Reply:
`map<type[]>` is allowed, as is `map<type[][]>`, `map<type>[]`, `map<type>[][]`, etc. `map<map<_>>` is the only thing that isn't allowed. We catch that in this framework in a sort-of hacky way by checking for that at `mro` printing time, and forcing it to just terminate the recursion by printing `map` instead of `map<map<...>>`